src/public/oauth-authorize.html

```diff
@@ -0,0 +1,108 @@
+<html>
+	<head>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width">
+
+		<title>Authorize application - Pokémon Showdown!</title>
+
+		<link rel="stylesheet" href="//pokemonshowdown.com/style/global.css">
+	</head>
+	<body>
+		<div class="body">
+		<header>
+			<div class="nav-wrapper"><ul class="nav">
+				<li><a class="button nav-first" href="/"><img src="/images/pokemonshowdownbeta.png" srcset="/images/pokemonshowdownbeta.png 1x, /images/pokemonshowdownbeta@2x.png 2x" alt="Pokémon Showdown" width="146" height="44"> Home</a></li>
+				<li><a class="button" href="/dex/">Pokédex</a></li>
+				<li><a class="button" href="//replay.pokemonshowdown.com/">Replays</a></li>
+				<li><a class="button" href="/ladder/">Ladder</a></li>
+				<li><a class="button nav-last" href="/forums/">Forum</a></li>
+				<li><a class="button greenbutton nav-first nav-last" href="//play.pokemonshowdown.com/">Play</a></li>
+			</ul></div>
+		</header>
+
+		<div class="main"><section class="section">
+			<h2>Authorize {{client}}</h2><hr />
+			<div class="infobox">
+				{{client}} by {{client_name}} wants to access your Pokemon Showdown account.<br />
+				This is public data only, and will allow you to log in on their site using your Pokemon Showdown account for two weeks.<br /><br /><hr />
+				<a id="auth" class="button greenbutton nav-first nav-last">Authorize</a><br /><br />
+				<small>Authorizing will redirect you to {{redirect_uri}}</small>
+			</div>
+		</section></div>
+	</div>
+
+	<footer>
+		<p>
+			<small><a href="/rules">Rules</a> | <a href="/privacy">Privacy policy</a> | <a href="/credits">Credits</a> | <a href="/contact">Contact</a></small>
+		</p>
+	</footer>
+
+	<script>
+		var safeJSON = function (callback) {
+			return function (data) {
+				if (data.length < 1) return;
+				if (data[0] == ']') data = data.substr(1);
+				return callback(JSON.parse(data));
+			};
+		};
+		var params = new URLSearchParams(location.search);
+		var tokenName = 'ps-' + params.get('client_id') + "-token";
+		var token = localStorage.getItem(tokenName);
+		if (token) {
+			params.set('token', token);
+		}
+		var acting = false;
+
+		function getAssertion() {
+			$.get('/api/oauth/api/getassertion', {
+				token: params.get('token'),
```

Comment:
Should we even bother making the request if we don't have a token?

Reply:
It's only ever called after the token is set, so there's no worry of that.